sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_client_async.py

```diff
@@ -897,13 +902,52 @@ async def _message_generator_async(self, timeout=None):
             if self._shutdown:
                 await self.close_async()
 
+    async def _on_disposition_received_async(self, message_delivery, reason, state):
+        message_delivery.reason = reason
+        if reason == LinkDeliverySettleReason.DISPOSITION_RECEIVED:
+            # pylint: disable=too-many-boolean-expressions
+            if state and (SEND_DISPOSITION_ACCEPT in state or
```

Comment:
I think I can also just check that error=None on the disposition? Thoughts?

Reply:
wont state only have error info if the disposition state is reject? As it is disposition won't have error otherwise

Reply:
The format of `state` is:
```
{'accepted': []} -> complete
or
{'released': []} -> release
or
{'rejected': [[b'com.microsoft:message-lock-lost', b'The.....', None]]} -> lock timeout
or
{'modified': [None, False, None]} -> abandon
or
{'modified': [None, True, None]} -> defer
or
{'rejected': [None]} -> deadletter
```

I don't know if "received" is expected from the service, I will check :)